arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Validated.kt

```diff
@@ -1152,6 +1216,23 @@ public inline fun <E, A> Validated<E, A>.findValid(SE: Semigroup<E>, that: () ->
     { Valid(it) }
   )
 
+/**
+ * Apply a function to a Valid value, returning a new Validation that may be valid or invalid
+ *
+ * Example:
+ * ```
+ * Valid(5).andThen { Valid(10) } // Result: Valid(10)
+ * Valid(5).andThen { Invalid(10) } // Result: Invalid(10)
+ * Invalid(5).andThen { Valid(10) } // Result: Invalid(5)
+ * ```
+ */
+public inline fun <E, A, B> Validated<E, A>.andThen(f: (A) -> Validated<E, B>): Validated<E, B> {
+  return when (this) {
+    is Validated.Valid -> f(value)
+    is Validated.Invalid -> this
+  }
+}
```

Comment:
Nit: Arrow codebase prefers expression bodies where easily possible.

```suggestion
public inline fun <E, A, B> Validated<E, A>.andThen(f: (A) -> Validated<E, B>): Validated<E, B> =
  when (this) {
    is Validated.Valid -> f(value)
    is Validated.Invalid -> this
  }
```